### DISCUSSION TITLE
Why does wingetcreate require read/write access to all public and private repositories when submitting a manifest?

### DISCUSSION THREAD

**Discussion by G0rocks** [microsoft/winget-create#33] [Q&A]
Yeah the title says it all, I don't know what you want me to write here ¯\_(ツ)_/¯

**Comment by ryfu-msft** [comment]
Hey @G0rocks, 
The entire submission flow/logic of WingetCreate goes as follows:
- Fork the winget-pkgs repo
- Create a new branch in your forked repository
- Push the generated manifests to the newly created branch
- Create a pull request

In order for WingetCreate to complete all of those steps on your behalf, we require read/write access.

As far as we know, this is the only way we can successfully perform all of those steps above to submit a PR on your behalf.
The tool is implemented as a GitHub app to connect to your repo, so if you know a better way to narrow the scope to only apply to a user's forked winget-pkgs repo, please let us know :)

**Reply by G0rocks** [reply]
All righty, that makes a lot of sense for the read/write access.

I'm sorry I don't have a better way to make the scope be only the forked repo and I didn't find out how to make the scope be only public repos either though I did search around and found [this](https://docs.github.com/en/developers/apps/managing-github-apps/editing-a-github-apps-permissions). Though I'm a total noob in github apps (never made one). Maybe this is something you could email github about and ask them ¯\_(ツ)_/¯

Thanks for the answer though :) Good luck!
